containers/proxy-salt-broker-image/prepare_target.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Moves to /target any file to be preserved in the slimmed-down image
+
+set -xe
+
+mkdir -p /target/etc/rhn
+mv /etc/rhn/rhn*.conf /target/etc/rhn/rhn.conf
+
+mkdir -p /target/etc/salt
+mv /etc/salt/broker /target/etc/salt
+
+mkdir -p /target/usr/share/rhn/config-defaults
+mv /usr/share/rhn/config-defaults/rhn*.conf /target/usr/share/rhn/config-defaults
+
+mkdir -p /target/usr/bin
+mv /usr/bin/salt-broker /target/usr/bin/
+mv /usr/bin/uyuni-configure.py /target/usr/bin/
+
+SITE_PACKAGES=usr/lib/$(ls /usr/lib/ | grep python)/site-packages
```

Comment:
Are you sure this will always return only one directory?

Reply:
It should, at least in the limited context of the container build.

Regardless, let me clarify the context in which I wrote this (admittedly in half-hacked way): this was an attempt to try multi-staged builds as a technique to reduce image size.

It was not meant to be the one solution to be promoted in general, or in this specific image case - just an experiment to see if it was viable especially in the context of the Open Build Service. The experiment itself succeeded in that it did not highlight any OBS-specific problem, but it did not necessarily give clarity on what to do with the other images. One clear disadvantage is that the script here might not be trivial and might turn out to be brittle.

So FMPOV it is still to be determined whether this is the preferred way to trim the image size. Adjusting packages is an alternative option, and forcibly removing packages during the image build is yet another one. Even removing files directly is possible. All these possibilities (and maybe others escaping my attention) should be evaluated when we look at image trimming.

Until that time, I suggest to either leave the code here alone or remove the multi staged build altogether if it helps with review clarity - reverting later is always easily possible.

HTH

CC @rjmateus

Reply:
In my opinion, at this stage, we should keep it as simple as possible, and tackle it when we got to the stage of reducing image size.
However, is this code is working now, and we don't expect any major problem with it, then just let it stay and it can be refactored latter.